### PR TITLE
 customize klusterlet search

### PR DESCRIPTION
**Pull request by xrajesh**
Signed-off-by: Xavier Dharmaiyan <xavier@redhat.com>

**Related Issue:**  stolostron/backlog#18606

### CODE REVIEW

File: addon/manifests/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -48,7 +64,15 @@ spec:
           value: {{ .Values.global.proxyConfig.NO_PROXY }}
         {{- end }}
         resources:
-          {{- toYaml .Values.resources | nindent 10 }}
+          requests:
+            memory: 128Mi
```
Comment (jlpadilla) [inline addon/manifests/chart/templates/deployment.yaml:68]: Can we also allow to customize the request. This may also cause problems in some scenarios.

Comment (xrajesh) [reply addon/manifests/chart/templates/deployment.yaml:68]: @jlpadilla Could you please explain the scenario - Request should be the minimum that we need to run the pod . And we want our users to get that.

Comment (jlpadilla) [reply addon/manifests/chart/templates/deployment.yaml:68]: `request` is the minimum amount or memory that kubernetes will **guarantee**.  If the pod is using more than the request, the kubernetes scheduler may kill the pod to assign it to another workload with a higher priority.

In some large clusters, users may need to change the request to ensure that the pod gets the necessary resources.

Comment (andreadecorte) [reply addon/manifests/chart/templates/deployment.yaml:68]: request should be set to the real avg usage of your pod, if you declare 128Mi but you are using 1Gi the scheduler won't be able to assign properly resources and you can cause evictions. So this should be definitely customizable

Comment (xrajesh) [reply addon/manifests/chart/templates/deployment.yaml:72]: Added support for request memory. @andreadecorte @jlpadilla

Comment (jlpadilla) [reply addon/manifests/chart/templates/deployment.yaml:72]: thanks

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Pull request merged**